src/Middleware/BaseMiddleware.php

```diff
@@ -37,12 +37,12 @@ public function __construct(JWTAuth $auth)
      *
      * @param   \Illuminate\Http\Request $request
      *
-     * @throws \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
+     * @throws \Tymon\JWTAuth\Exceptions\JWTException
      */
     public function checkForToken(Request $request)
     {
         if (! $this->auth->parser()->setRequest($request)->hasToken()) {
-            throw new BadRequestHttpException('Token not provided');
+            throw new JWTException('Token not provided');
```

Comment:
I don't think this is the same as #407, because this is within middleware and a http specific exception is relevant here imo